.circleci/config.yml

```diff
@@ -101,7 +101,8 @@ jobs:
       - run: make test TEST_FLAGS="-race -tags integration -timeout 5m"
       - run: make check-generated
       - run: make all
-      - run: E2E_KIND_CLUSTER_NUM=4 make e2e
+      - run: E2E_KIND_CLUSTER_NUM=4 CACHE_BACKEND=memcached make e2e
+      - run: E2E_KIND_CLUSTER_NUM=4 CACHE_BACKEND=redis make e2e
```

Comment:
This is overkill because tests take more than 10 minutes and only some of them use the cache (those with --registry-disable-scanning=false)

Reply:
what about parallelize?

Reply:
@2opremio I guess we can achieve parallelism using the _workflow_ section of CircleCI, please take a look [here](https://github.com/fluxcd/flux/pull/2815/files#diff-1d37e48f9ceff6d8030570cd36286a61R207-R209).

Actually, tests are running: let's see how they end up.

Reply:
@2opremio in case you missed: I implemented the e2e tests for each image cache backing service, named `e2e-memcached` and `e2e-redis`: since they're parallel, we're not anymore wasting time during the CI.

Although, if you don't like this approach, we could abstract the `bats` test cases using a function to run all the assertions. However, I feel this solution a bit tricky because it looks more complex for those who never contributed.

Waiting for your feedback, I'd really like to get it merged :)

Reply:
I still think it's overkill because most of the e2e tests run without using a cache (because we run them with `--registry-disable-scan`). I would much rather have duplicate runs only for the tests using the cache.